module-with-tests/src/main/java/org/apache/flink/connector/testing/SomeClass.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.testing;
+
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+/** A dummy class; only exists so that the compile/packaging plugins have something to do. */
+public class SomeClass {
+    public static void main(String[] args) {
+        // this call is an intended (for testing) violation of the
+        // ConnectorRules#CONNECTOR_CLASSES_ONLY_DEPEND_ON_PUBLIC_API archunit rule that was fixed
```

Comment:
Actually, I'm a bit lost.  Does this fail the build or not? What prevents it from failing if the architecture tests are enabled? We always build against the same Flink version after all.

Reply:
It was to simulate a rule recently changed in flink-snapshot (the initial problem that triggered this PR). When the archunit tests are enabled, if the violation is added to the store (via freeze) it should not fail the build and it should fail the build otherwise. But anyway as the general case is to disable archunit for both snapshots and non-main flink versions, I think it is better to remove the simulated violation indeed.

Reply:
Done